eclair-core/src/test/scala/fr/acinq/eclair/integration/MessageIntegrationSpec.scala

```diff
@@ -65,6 +71,7 @@ class MessageIntegrationSpec extends IntegrationSpec {
     val probe = TestProbe()
     val eventListener = TestProbe()
     nodes("B").system.eventStream.subscribe(eventListener.ref, classOf[OnionMessages.ReceiveMessage])
+    sleep(100 millis)
```

Comment:
this would work, another approach would be to test that the subscription is active: create a dummy  `OnionMessages.ReceiveMessage` and publish it in a loop until `eventListener.reveiceOne() == dummy`

Reply:
I agree with sstone that we need something more robust.
Nothing guarantees you than 100 milliseconds is enough, this is an arbitrary value.
It would be much better to find a business logic condition to test for, that will guarantee that your nodes are in the right state.

The trick that sstone shared is one we've already use in a few places (but I don't remember where exactly).